Update context.mdx (#813)
solidjs/solid-docs: src/routes/concepts/context.mdx

When you have a large [component tree](/concepts/components/basics#component-trees) that requires state to be shared, context can be used.
Context can be employed to avoid [prop drilling](/concepts/components/props#prop-drilling), which is the practice of passing props through intermediate elements without using them directly.

> If you only want to avoid passing "some" props through a "few" layers, adjusting your component hierarchy may be an easier solution, when applicable.

Reviewer: ```suggestion
If you want to avoid passing some props through a few layers, when applicable, adjusting your component hierarchy may be an easier solution.
```

> [Signals](/concepts/signals), when injected directly into a component that needs them, will always be the simplest and most explicit solution.

Reviewer: ```suggestion
[Signals](/concepts/signals) are often the simplest solution since they can be imported directly into the components that need them.
```

> Context, on the other hand, is designed to share data that can be considered public or "global", i.e. information that is regularly accessed by many components in your application's component tree.

Reviewer: ```suggestion
Context, however, is designed to share data that is global to an application or for information that is regularly accessed by multiple components in an application's component tree.
```

> It offers a way to access state across the hierarchy without passing props through intermediate layers and injecting them directly into components.

Reviewer: ```suggestion
This offers a way to access state across an application without passing props through intermediate layers or importing them directly into components.
```
